### ISSUE TITLE
Library lookup fails on Debian 10.4

### ISSUE THREAD

**Issue by pascalito** [caterinaurban/apronpy#1]
On Debian 10.4, with regular package `libapron` installed, the existing file `/usr/lib/libpolkaMPQ.so` is not found by the following line of `cdll.py`:
``
libpolkaMPQ` = CDLL(util.find_library('libpolkaMPQ.so'))
``
but is found if the line is changed to:
``
libpolkaMPQ = CDLL(util.find_library('polkaMPQ'))
``

I guess that would work with other systems too. It would also be more coherent with the way that the `libc` and `libgmp` are looked up.

**Comment by caterinaurban** [comment]
Oops, thanks, side-effect of developing and testing this on a Mac only. Indeed, your fix does not work on a Mac, I have to add a switch that looks up in the right way depending on the system.

**Comment by caterinaurban** [comment]
I tested the fix on Ubuntu 20.04 LTS. Let me know if this does not work for you!

**Comment by pascalito** [comment]
Almost fixed. _The library lookup now works._
Unfortunately, Taylor1+ (i.e. `libt1pD.so`, `libt1pMPQ.so` and `libt1pMPFR.so`) is currently not bundled in the [Debian 10 libapron package](https://packages.debian.org/buster/libapron). That triggers an exception when loading `manager.py` preventing any use of `apronpy`. 
Supporting the absence of Taylor1+ can be fixed by a minor modification of `manager.py`. I _can propose you a merge request_ it if you allow me push. This can also be seen as an issue in the Debian package, so it is up to you.

**Comment by caterinaurban** [comment]
I suppose I could just move the managers of different domains into different files. That should solve the problem, no? Did you have a different suggestion?

**Comment by caterinaurban** [comment]
Ok, I moved the domain-specific managers (see 8bee1a83c01f19b81e0f1e4d5f852c341e7d4c43). Now if one does not use the Taylor1+ domain, it shouldn't need to loopkup its library.

**Comment by caterinaurban** [comment]
Excellent! Thanks ^^

**Comment by caterinaurban** [comment]
I simplified the library lookup (see 5c9381a829443291ce59d07d691de7fd75913379). Can you check if that still works for you? It seems to work on Debian 9.9. Thanks!

**Comment by pascalito** [comment]
Hello Caterina. The new library lookup fails in absence of Taylor1+, unfortunately.
```
[...]
  File "/home/sotin/depots/traces/apronpy/apronpy/cdll.py", line 12, in <module>
    libt1pD = CDLL('libt1pD.so')
  File "/usr/lib/python3.7/ctypes/__init__.py", line 356, in __init__
    self._handle = _dlopen(self._name, mode)
OSError: libt1pD.so: cannot open shared object file: No such file or directory
```
I do not know why Taylor1+ is not bundled in `libapron-dev` under Debian 10, nor if it was part of it in Debian 9.*.

**Comment by caterinaurban** [comment]
Thanks for checking! This is so annoying 👎  
I moved some things around (see 8a7e08e6929beeeeb97a9da648499be8c5d18bff). Now the lookup for Taylor1+ does not happen unless one actually wants to use the domain (I did the same thing as for the managers, basically). Hopefully, this fixes it for good! With the previous fix I had trouble on Debian 9.9 because `util.find_library()` was not finding the libraries...
